### ISSUE TITLE
Kelpie shield HUD gauge no longer displays properly

### ISSUE THREAD

**Issue by Goober5000** [scp-fs2open/fs2open.github.com#4148]
The GTF Kelpie's shield gauge displays as a white square:

![screen0144](https://user-images.githubusercontent.com/1878523/162057943-7782e41d-4b7f-4074-b219-c01495bae1d1.png)

Attached is a simple test mod that assigns this shield gauge to the Ulysses.

[kelpie_test.zip](https://github.com/scp-fs2open/fs2open.github.com/files/8430108/kelpie_test.zip)

**Comment by Baezon** [comment]
For anyone else, I've had a look around but as far as I can tell FSO is loading and using the .ani just fine. There's something about the image itself that's causing it to display wrong, but that goes deeper into bmpman than I have familiarity with.

**Comment by Goober5000** [comment]
Yep, this is indeed a problem with the shield gauge file itself, not with the code.  The palette indexed colors were in opposite order from other palettes that do work.  Surprisingly, there is a dialog in GIMP to edit this, so I reversed the order, rebuilt the ANI, and it works now.

Here is the fixed file:
[kelpie-shld.zip](https://github.com/scp-fs2open/fs2open.github.com/files/8439149/kelpie-shld.zip)

So now the question becomes: is an oppositely-ordered palette something we want to handle?

**Comment by Goober5000** [comment]
Having looked at the ani palette code a bit, I don't think there is any way to tell.  But I'll ping @notimaginative in case he has any ideas.

**Comment by notimaginative** [comment]
I need to have a look at that code since it's been so long, but my first instinct is that this is a code bug. Each pixel is just an index into the palette so the order really shouldn't matter. And both GIMP and aniviewer handle it just fine so I don't think there is any issue with the ani itself.

I'll go through the code when I get the chance and see what I can find.

**Comment by notimaginative** [comment]
Ah, looks like an aabitmap requires a very specific palette and is hard coded to that layout. We actually could detect if that layout is reversed but I don't think it's worth the extra code to deal with it. Easier to just make sure the palette for aabitmaps is documented properly so that all such ani/pcx are created as the code expects.

Per the code comments about the aabitmap palette:
```
// 8 bit-per-pixel aa bitmaps are a bit special since they only use a palette index value in the range [0, 15]. These 
// palette indexes must be remapped to alpha values between [0, 255] which is what graphics code expects. Palette 
// range [0, 14] is a gradient from black to white, and palette index 15 is a special color which indicates the background
// area of a HUD gauge. Retail code uses the final alpha value for index 1 for this special index to give gauges a dark
// transparent background.
```

**Comment by Goober5000** [comment]
Interesting.

**Comment by Goober5000** [comment]
@notimaginative Do any documentation changes need to be made in the codebase?  (Or anywhere else for that matter)

**Comment by notimaginative** [comment]
Not sure really. Plus, ANI aabitmaps are handled slightly differently than PCX aabitmaps. The comment I posted earlier is for the ANI version (`packunpack.cpp:154`) which is for the hud graphics. The PCX version (`pcxutils.cpp:315`) has a similar but different decoding for the same palette range.

So I don't know whether one or both of those cases need to be better documented than just the two code comments. The wiki is pretty vague about the palette for hud gauges, mostly just saying to make sure it matches retail anis or it will render wrong. So I guess at least the ANI palette could go somewhere in there to detail what it should actually look like.

**Comment by MjnMixael** [comment]
I think the actionable step here is just to comment and document the palette? Correct me if I'm wrong.

**Comment by Goober5000** [comment]
Agreed.  This isn't really my area of expertise, but Claude had some ideas.  I'll post a PR.

**Comment by Goober5000** [comment]
Resolved by #7410
